### PR TITLE
add method for custom exiftool commands and getting the raw output fr…

### PR DESCRIPTION
**Pull request by KevinClauson**
This will allow a user to create their own exiftool command and get the raw output from exiftool.  I was interested in getting the family name of the tag but that is not supported as well as getting a response of a json string.  So for example this pr will allow a command like:

exiftool -a -u -g1 -j

This Exif Command -a is for all metadata so includes duplicate tags -u is to include any unknown tags -g1 sorts the tags by group for family 1 and -j exports in json.

This way we have the family tag and some family tag can have same tag name, so keeps everything separate.  So can use the fantastic functionality of stay_open, but have greater flexibility in the commands we can run.

This would seem to address 3 of the lodged issues:
https://github.com/mjeanroy/exiftool/issues/289
https://github.com/mjeanroy/exiftool/issues/117
https://github.com/mjeanroy/exiftool/issues/247
As well as allowing a workaround for this issue and others like it that want to better control the Exiftool options when getting metadata:
https://github.com/mjeanroy/exiftool/issues/442

### CODE REVIEW

**Review by di-ruan — APPROVED** [review]

**Comment by mjeanroy** [comment]
Hi,
Thanks for the PR @KevinClauson!

Just to let you know, I'm currently in vacation and won't check it until 1 or 2 weeks, but it's already in my TODO when I'm back!

**Comment by KevinClauson** [comment]
Thanks @mjeanroy, I figured you were on vacation due to the time of year and your commit history.  The PR was actually the last thing I did before going on my own vacation, which I'm just getting back from.  First full day of work tomorrow.  Definitely don't interrupt your vacation on account of this PR.  Once you get back in the saddle, it will be great to get your thoughts.  Hope you enjoy your vacation.

**Comment by KevinClauson** [comment]
Hi @mjeanroy, hope you had a pleasant vacation.  Just wanted to touch base on the above PR and get your thoughts on whether this is something you would want to merge.

**Comment by KevinClauson** [comment]
Hey @mjeanroy just wanted to check in on this.

**Comment by KevinClauson** [comment]
Should add, no worries if you don't want to merge this.  I would be interested on your reasoning though, as I will be wanting to use something like this in production.

**Comment by mjeanroy** [comment]
@KevinClauson No, it's definitely interesting! Sorry for the delay, I just have too much work since a few weeks.

**Comment by mjeanroy** [comment]
@KevinClauson Could you reformat to use tab for indentation so it's consistent with the codebase?

File: src/main/java/com/thebuzzmedia/exiftool/ExifTool.java [hunk 1]
```diff
@@ -525,6 +526,41 @@ private Map<Tag, String> getImageMeta(File image, Collection<? extends Tag> tags
...
+		List<String> args = new ArrayList<>(expectedSize);
+
+		addAll(args, arguments);
+
+		// Add image argument.
+		args.add(image.getAbsolutePath());
+
+		// Add last argument.
+		// This argument will only be used by exiftool if stay_open flag has been set.
+		args.add("-execute");
```
Comment (mjeanroy) [inline src/main/java/com/thebuzzmedia/exiftool/ExifTool.java:556]: What happends if `-execute` is already in the `arguments` list?

Comment (KevinClauson) [reply src/main/java/com/thebuzzmedia/exiftool/ExifTool.java]: This is a very good question and it got me thinking it this wasn't limiting the scope of the commands a user could enter.  And you are right why be adding "-execute" when the whole idea is to expose the ability for a user to enter their own exiftool command.   ok so I updated the PR and updated the tests.

I took out passing in the image file.  So this way if I want a command where I use exiftool to copy certain metadata from one image to to another I can use this method.  

What do you think about this?

Comment (KevinClauson) [reply src/main/java/com/thebuzzmedia/exiftool/ExifTool.java]: @mjeanroy just wanted to check back in on this.

Comment (mjeanroy) [reply src/main/java/com/thebuzzmedia/exiftool/ExifTool.java]: Make sense, so this function is more a helper function on top of this exiftool wrapper :)

**Comment by mjeanroy** [comment]
Version 3.1.0 has just been published with this change, thanks @KevinClauson and very sorry for the unexpected delay!

**Pull request merged**